oggm/core/models/flowline.py

```diff
@@ -623,6 +630,154 @@ def step(self, dt=SEC_IN_MONTH):
         # Next step
         self.t += dt
 
+class MUSCLSuperBeeModel(FlowlineModel):
+    """This model is based on Jarosch et al. 2013 (doi:10.5194/tc-7-229-2013)
+       The equation references in the comments refer to the paper for clarity
+    """
+    def __init__(self, flowlines, mb_model=None, y0=0., Aglen=None, fs=None, fd=None,
+                 fixed_dt=None, min_dt=SEC_IN_DAY, max_dt=SEC_IN_MONTH):
+
+        """ Instanciate.
+
+        Parameters
+        ----------
+
+        Properties
+        ----------
+        #TODO: document properties
+        """
+
+        if len(flowlines) > 1:
+            raise ValueError('MUSCL SuperBee model does not work with tributaries.')
+
+        super(MUSCLSuperBeeModel, self).__init__(flowlines, mb_model=mb_model,
+                                            y0=y0, Aglen=Aglen, fs=fs, fd=fd)
+        self.dt_warning = False,
+        if fixed_dt is not None:
+            min_dt = fixed_dt
+            max_dt = fixed_dt
+        self.min_dt = min_dt
+        self.max_dt = max_dt
+    
+        
+    def phi(self, r):
+        """ a definition of the limiting scheme to use"""
+        # minmod limiter Eq. 28
+        # val_phi = numpy.maximum(0,numpy.minimum(1,r))
+        
+        # superbee limiter Eq. 29
+        val_phi = np.maximum(0,np.minimum(2*r,1),np.minimum(r,2))
+        
+        return val_phi
+
+    def step(self, dt=SEC_IN_MONTH):
+        """Advance one step."""
+
+        # This is to guarantee a precise arrival on a specific date if asked
+        min_dt = dt if dt < self.min_dt else self.min_dt
+        dt = np.clip(dt, min_dt, self.max_dt)
+
+        fl = self.fls[0]
+        dx = fl.dx_meter
+        width = fl.widths_m
+        
+        """ Switch to the notation from the MUSCL_1D example
+            This is useful to ensure that the MUSCL-SuperBee code
+            is working as it has been benchmarked many time"""
+         
+
+        # mass balance
+        m_dot = self.mb.get_mb(fl.surface_h, self.yr)
```

Comment:
You asked last time round if mb should not be multiplied by glacier with here. I don't know. I took your Karthaus model as a template and there (line 593 above) you don't do that either. Please elaborate on your thought